db/view/view.cc

```diff
@@ -1772,15 +1774,36 @@ get_view_natural_endpoint(
         }
     }
 
-    SCYLLA_ASSERT(base_endpoints.size() == view_endpoints.size());
     auto base_it = std::find(base_endpoints.begin(), base_endpoints.end(), me);
     if (base_it == base_endpoints.end()) {
         // This node is not a base replica of this key, so we return empty
         // FIXME: This case shouldn't happen, and if it happens, a view update
         // would be lost.
         ++cf_stats.total_view_updates_on_wrong_node;
-        return {};
+        return std::make_pair(std::nullopt, std::nullopt);
+    }
+
+    // The following code handles the case where after a replication factor change
+    // the base replica completed the corresponding tablet migration at a different
+    // time than the view replica.
+    // RF can be changed only by 1 at a time, so the difference between the number
+    // of base and view replicas should be at most 1.
+    SCYLLA_ASSERT(std::abs((int)(base_endpoints.size() - view_endpoints.size())) <= 1);
```

Comment:
Not sure if it's that simple. There can be many DCs with different RFs. If we change e.g. RF from 2 to 3 in one DC, do the same in the 2nd DC and do the opposite, i.e. change RF from 3 to 2, in the 3rd DC, we end up satisfying (?) this condition, however there would be 3 pending migrations.

Reply:
The `base_endpoints` and `view_endpoints` are filtered by DC of the current replica so I don't think we should get any interference from other DC's here. But I can add this consideration to the comment.

I also looked at `test_multidc_alter_tablets_rf` that you mention in another comment and I see that when trying to modify RF of multiple DCs we expect to get a `Only one DC's RF can be changed at a time and not by more than 1` error - doesn't this prevent the scenario that you mention?

Reply:
> doesn't this prevent the scenario that you mention?

Possibly. I'm rather asking questions here than giving definitive remarks. Maybe it'd be better to make this assertion not rely on the ALTER KS code, but assert the same as what ALTER asserts. Just something to consider.

Reply:
Ok, I looked at the ALTER KS code and it looks like it follows the `Only one DC's RF can be changed at a time and not by more than 1` message. Though the changes in other DCs really shouldn't matter here - we pair the replicas only after filtering out other DCs from the list.